pkg/controller/backup_repository_controller.go

```diff
@@ -165,7 +165,7 @@ func (r *BackupRepoReconciler) needInvalidBackupRepo(oldObj client.Object, newOb
 			"old bucket": oldStorage.Bucket,
 			"new bucket": newStorage.Bucket,
 		}).Info("BSL's bucket has changed, invalid backup repositories")
-
+		r.invalidateBackupReposForBSL(context.TODO(), oldObj)
```

Comment:
For the existing logics:

- For each updated BSL, `needInvalidBackupRepo` is called
- Once  `needInvalidBackupRepo` returns true, `invalidateBackupReposForBSL` is called and the repo is set as `notReady`

So could you explain why these changes are necessary?